### ISSUE TITLE
Decorators not allowed in JS even with 'experimentalDecorators' set

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#8905]
1. Create a new folder.
2. Add a `tsconfig.json`/`jsconfig.json`.
3. Add `"experimentalDecorators": true`.
4. Create a new `.js` file that has a decorated class.

Expected: No error.
Actual: Error that you need to add `"experimentalDecorators": true`.

**Comment by billti** [comment]
Hmmm, isn't this exactly what https://github.com/Microsoft/TypeScript/pull/6881 fixed? Maybe a regression (though hard to say, as you didn't mention what build/branch you are using :-p ).

**Comment by billti** [comment]
p.s. Just tried it with a build from a couple days ago in `master`, and it works fine with the below `tsconfig.json`. It gives the error if I change the setting to `false` (as expected). Please provide a repro (and details on what branch/build) if you are sure you are seeing issues.

```javascript
{
    "compilerOptions": {
        "experimentalDecorators": true,
        "allowJs": true
    }
}
```

**Comment by mhegazy** [comment]
i see this on latest too.

**Comment by DanielRosenwasser** [comment]
@mhegazy you see the behavior I'm reporting, or are you also saying it works for you?

@jramsay saw the same issue a few days ago as well. I used the nightly, so `Version 1.9.0-dev.20160525-1.0`.

**Comment by zhengbli** [comment]
Cannot repro this either. Talked to @mhegazy, turns out he had an invalid tsconfig when he saw this. Close until a repro can be found.

**Comment by akanieski** [comment]
I'm working on a TS project that's using gulp (+ aurelia-cli). Even with `experimentalDecorators: true` I still get red squigglies in VSCode.

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "sourceMap": true,
    "target": "es5",
    "module": "amd",
    "declaration": false,
    "noImplicitAny": false,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "lib": ["es2015", "dom"]
  },
  "exclude": [
    "node_modules"
  ],
  "filesGlob": [
    "./src/**/*.ts",
    "./test/**/*.ts",
    "./typings/index.d.ts",
    "./custom_typings/**/*.d.ts"
  ],
  "atom": {
    "rewriteTsconfig": false
  }
}
```
<img width="686" alt="decorators" src="https://cloud.githubusercontent.com/assets/595427/17084019/d64f4aa8-5180-11e6-9b3f-49c0efb537f3.png">

**Comment by akanieski** [comment]
I ran into this when I open a root folder that contains TS project for client and TS project for server (File-> Open Folder ..  /root ). The root has no tsconfig file. VSCode does not respect the tsconfig in each respective project folder ( /root/server/tsconfig.json  and  /root/client/tsconfig.json ). It is expecting a tsconfig to be in the root of the folder opened in VSCode ( /root/tsconfig.json ).

When I open another instance of VSCode and open the  /root/client  or  /root/server  folder the decorator warning goes away, as it should.

Is this by design? I'm not sure i'd expect it to find the nearest tsconfig.json in local folder or above.

I hope I'm making sense :-P

**Comment by zhengbli** [comment]
@akanieski it shouldn't be the case, the tsconfig file in the same folder should be respected regardless where you open as root in VSCode. Can you provide a small sample of the issue for reproing? I tried on my machine but I didn't see it happen. Also with your tsconfig.json content, I didn't get the error for the decorators. What version of VSCode / TypeScript are you using?

**Comment by akanieski** [comment]
Sure. Checkout the following repo:

https://github.com/akanieski/decorator-issue

<img width="275" alt="snip_20160729200708" src="https://cloud.githubusercontent.com/assets/595427/17266591/156583d8-55c8-11e6-8a00-9a3c188ce729.png">

**Comment by akanieski** [comment]
TS Version 2.1.0-dev.20160724

**Comment by zhengbli** [comment]
That doesn't seem to be the case for me:
![capture](https://cloud.githubusercontent.com/assets/1171301/17339471/e93cd000-58a0-11e6-92ce-775762ec4bd8.PNG)

Did you do anything else other than open the files, like switch git branches etc.?
I am using the same TS version.

**Comment by akanieski** [comment]
Nope. Let me try from another pc. Perhaps I have something interfering.

**Comment by juliusl** [comment]
Not sure if this helps but.. if I remove this line: 

` "lib": ["es2015", "dom"] `

reopen the file and wait a second the error goes away.

**Comment by jorgeas80** [comment]
Having this problem right now. VSCode 1.5.3 with Typescript 2.0.3. Running in a Ubuntu 16.04 box. Tried all the suggestions in this thread.

My tsconfig.json

```
{
  "compilerOptions": {
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "lib": ["es2015", "dom"],
    "mapRoot": "./",
    "module": "es2015",
    "moduleResolution": "node",
    "outDir": "../dist/out-tsc",
    "sourceMap": true,
    "target": "es5",
    "typeRoots": [
      "../node_modules/@types"
    ]
  }
}
```

**Comment by robinchesterman** [comment]
I had this problem in one project, but not in another, both using TS 2.0.3 and VS Code 1.5.3. The tsconfig that wasn't working (ignoring filesGlob and exclude):

`{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "sourceMap": true,
    "moduleResolution": "node",
    "allowJs": true
  }
}`

And the one that does: 

`  "compileOnSave": false,
  "compilerOptions": {
    "sourceMap": true,
    "target": "es5",
    "module": "amd",
    "declaration": false,
    "noImplicitAny": false,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "lib": ["es2015", "dom"]
  },`

I copied over the one that worked into the other project, and they both work. I tried swapping out every single option looking for a hint has to what might stop the experimentalDecorators flag being honoured, but with no luck. So I have it working, but I have no idea why.

**Comment by billti** [comment]
You have `allowJs` to true in the first one. This should be giving you an error when you try to compile that any .js files will overwrite themselves. You'd need to either specify `"noEmit": true` or `"outDir": "../someOtherFolderNotInTheCompilation"` for this to be a valid configuration.

**Comment by anhtuank7c** [comment]
@billti Thank. It's work for me.

This is `tsconfig.json` in my React Native project:
```
{
    "compilerOptions": {
        "experimentalDecorators": true,
        "allowJs": true
    }
}
```

**Comment by cornelius-k** [comment]
In atom I tried @anhtuank7c suggestion and replaced my tsconfig with 

''''
{
    "compilerOptions": {
        "experimentalDecorators": true,
        "allowJs": true
    }
}
''''

which fixed the issue, then I tried reverting to my original config that I need that was causing the issue and the issue no longer occurs. Great!

**Comment by omencat** [comment]
Confirmed the two compilerOptions settings resolved the warnings. My original tsconfig.json file had those two settings, but a bunch of other settings. After adding settings back one at a time, It appears the conflict was the following:

```
  "include": [
    "*.ts",
    "src/**/*.ts"
  ],
```

I ended up replacing the include with exclude instead.

```
  "exclude": [
    "node_modules"
  ]
```

**Comment by dewwwald** [comment]
Someone already said this but when I added  `"lib": ["es2015", "dom"]` to the top level of the config file it started working / something stopped working.

**Comment by bofcarbon1** [comment]
jsconfig.json

{
    "compilerOptions": {
        "experimentalDecorators": true        
    }
}

When using VS Code in the root directory of your project and the warnings go away.
In my case I'm learning React and there aren't that many TypeScript tutorial versions 
around.

**Comment by steph643** [comment]
Same issue here. I've tried the solutions proposed above, but the warning is still there (VSCode 1.21.1 under Windows). 

My `jsconfig.json` file:
```
{
	"compilerOptions": { 
		"experimentalDecorators": true
	},
	"exclude": [
		"node_modules",
		"**/node_modules/*"
	]
}
```

Same thing with a `tsconfig.json` file:
```
{
	"compilerOptions": { 
		"experimentalDecorators": true,		
		"allowJs": true
	},
	"exclude": [
		"node_modules",
		"**/node_modules/*"
	]
}
```

**Comment by josecolella** [comment]
I followed the  solutions but I'm still getting a `Declaration expected` warning from vscode.
Here is a snippet from the compilerOptions of my tsconfig.json
` 
"compilerOptions": {
    "noImplicitAny": true,
    "removeComments": true,
    "preserveConstEnums": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "allowJs": true,
`
